pyqtgraph/functions.py

```diff
@@ -1483,20 +1483,36 @@ def arrayToQPath(x, y, connect='all'):
 
     path = QtGui.QPainterPath()
 
-    #profiler = debug.Profiler()
     n = x.shape[0]
+
     # create empty array, pad with extra space on either end
     arr = np.empty(n+2, dtype=[('c', '>i4'), ('x', '>f8'), ('y', '>f8')])
+
     # write first two integers
-    #profiler('allocate empty')
     byteview = arr.view(dtype=np.ubyte)
     byteview[:16] = 0
     byteview.data[16:20] = struct.pack('>i', n)
-    #profiler('pack header')
+
     # Fill array with vertex values
     arr[1:-1]['x'] = x
     arr[1:-1]['y'] = y
 
+    # inf/nans completely prevent the plot from being displayed starting on 
+    # Qt version 5.12.3; these must now be manually cleaned out.
+    isfinite = None
+    qtver = [int(x) for x in QtVersion.split('.')]
+    if qtver >= [5, 12, 3]:
+        isfinite = np.isfinite(x) & np.isfinite(y)
+        if not np.all(isfinite):
+            # credit: Divakar https://stackoverflow.com/a/41191127/643629
+            mask = ~isfinite
+            idx = np.arange(len(x))
+            idx[mask] = -1
+            np.maximum.accumulate(idx, out=idx)
+            first = np.searchsorted(idx, 0)
+            idx[:first] = first
+            arr[1:-1] = arr[1:-1][idx]
```

Comment:
```suggestion
            if first < len(x):
                # Replace all non-finite entries from beginning of arr with the first finite one
                idx[:first] = first
                arr[1:-1] = arr[1:-1][idx]
```
Fixes the case where the whole buffer contains non-finite items in x or y (or both).